code/modules/lighting/lighting_system.dm

```diff
@@ -246,10 +246,27 @@
 	anchored = 1
 
 /atom/movable/light/Destroy(force)
-	if(force)
-		. = ..()
-	else
+	if(!force)
 		return QDEL_HINT_LETMELIVE
+	var/turf/T = loc
+	. = ..()
+	if (T.lighting_object == src)
+		T.lighting_object = null
+	
+/atom/movable/light/New()
+	if (!loc || !isturf(loc))
```

Comment:
just ```if(!isturf(loc))```

if it's not a turf, it succeeds
if it's null, it succeeds

same results, one less condition.